invenio_app_ils/locations/loaders/jsonschemas/location.py

```diff
@@ -21,3 +66,37 @@ class LocationSchemaV1(RecordMetadataSchemaJSONV1):
     email = fields.Email()
     phone = fields.Str()
     notes = fields.Str()
+    opening_weekdays = fields.List(fields.Nested(OpeningWeekdaySchema))
+    opening_exceptions = fields.List(fields.Nested(OpeningExceptionSchema))
+
+    @post_load
+    def postload_checks(self, data, **kwargs):
+        """Sort exceptions and validate record."""
+        weekdays = data["opening_weekdays"]
+        new_weekdays = [None for _ in _weekday_names]
+
+        exists_open = False
+        for weekday in weekdays:
+            name = weekday["weekday"]
+            exists_open = exists_open or weekday["is_open"]
+            index = _weekday_indices[name]
+            if new_weekdays[index]:
+                raise ValidationError("Duplicate weekday.", field_names=["opening_weekdays"])
+            new_weekdays[index] = weekday
+        if len(weekdays) != len(new_weekdays):
+            raise ValidationError("Missing weekdays.", field_names=["opening_weekdays"])
+        if not exists_open:
+            raise ValidationError("At least one weekday must be declared as open.",
+                                  field_names=["opening_weekdays"])
+
+        exceptions = data["opening_exceptions"]
+        exceptions.sort(key=lambda ex: ex["start_date"])
+        previous = None
+        for exception in exceptions:
+            if previous:
+                if previous["end_date"] >= exception["start_date"]:
+                    raise ValidationError("Exceptions must not overlap.",
+                                          field_names=["opening_exceptions"])
+            previous = exception
+
```

Comment:
here you could add an extra check, since you have the record in the context with the previous exceptions and the new exceptions and weekdays to check if there is at least a day open in the next 365 days, if you want (see my comment before for the while True).

Reply:
(won't fix)